### PR TITLE
5934 Replace blocks with bricks in types

### PR DESCRIPTION
**Pull request by turbochef**
## What does this PR do?

- Part of #5934 
- `BlockDefinition` > `BrickDefinition`
- `BlockIcon` > `BrickIcon`
- `BlockConfig` > `BrickConfig`
- `IExtensionPoint` > `StarterBrick`
- `Block` > `BrickABC`
- `BlockType` > `BrickType`
- `ResolvedBlockConfig` > `ResolvedBrickConfig`
- `BlockArgs` > `BrickArgs`
- `BlocksRegistry` > `BricksRegistry`
- `BlockArgsContext` > `BrickArgsContext`

- blockFactory.ts > brickFactory.ts
- useBlockOptions.ts > useBrickOptions.ts
- blockFactories.ts > brickFactories.ts

I likely have a few more but there are already so many changes I wanted to cut these off

## Checklist

- [ ] Add tests
- [x] Designate a primary reviewer: @mnholtz

### CODE REVIEW

File: src/blocks/readers/jquery.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ type CommonSelector = ChildrenSelector | SingleSelector;
 type Selector = CommonSelector & {
   selector?: string;
 
-  // Block until the element is available
+  // Brick until the element is available
```
Comment (mnholtz) [inline src/blocks/readers/jquery.ts:49]: Revert this, it's the verb "block" not a `Block`

File: src/types/brickTypes.ts [hunk 2]
```diff
@@ -15,19 +15,19 @@
...
 import { validateRegistryId } from "@/types/helpers";
 import { type Schema, type UiSchema } from "@/types/schemaTypes";
-import { type BlockArgs, type BlockOptions } from "@/types/runtimeTypes";
+import { type BrickArgs, type BrickOptions } from "@/types/runtimeTypes";
 import { type RegistryId, type Metadata } from "@/types/registryTypes";
-import { type BlockIcon } from "@/types/iconTypes";
+import { type BrickIcon } from "@/types/iconTypes";
 
 /**
  * An instance of a re-usable block.
```
Comment (mnholtz) [inline src/types/brickTypes.ts:27]:
```suggestion
 * An instance of a re-usable brick.
```

File: src/types/brickTypes.ts [hunk 3]
```diff
@@ -15,19 +15,19 @@
...
- * @see BlockDefinition
+ * @see BrickDefinition
  */
-export interface IBlock extends Metadata {
+export interface Brick extends Metadata {
   /** A JSON schema of the inputs for the block */
```
Comment (mnholtz) [inline src/types/brickTypes.ts:31]:
```suggestion
  /** A JSON schema of the inputs for the brick */
```

File: src/types/brickTypes.ts [hunk 4]
```diff
@@ -87,7 +87,7 @@ export interface IBlock extends Metadata {
    * Defined as a promise to support blocks that refer to other blocks (and therefore need to look up the status of
```
Comment (mnholtz) [inline src/types/brickTypes.ts:87]:
```suggestion
   * Defined as a promise to support bricks that refer to other bricks (and therefore need to look up the status of
```

File: src/types/brickTypes.ts [hunk 5]
```diff
@@ -87,7 +87,7 @@ export interface IBlock extends Metadata {
...
    * the other blocks to resolve their isRootAware status).
```
Comment (mnholtz) [inline src/types/brickTypes.ts:88]:
```suggestion
   * the other bricks to resolve their isRootAware status).
```

File: src/types/brickTypes.ts
Comment (mnholtz) [inline src/types/brickTypes.ts:1]: Theres a few usages of the word `block` in comments in this file. Let's replace those too (I added suggestions for a few, but didn't go through the rest).

File: src/runtime/runtimeUtils.ts [hunk 6]
```diff
@@ -152,13 +152,13 @@ export async function shouldRunBlock(
 
 /**
  * Select the root element (or document) for a block based on the current root and the block's rootMode
```
Comment (mnholtz) [inline src/runtime/runtimeUtils.ts:154]:
```suggestion
 * Select the root element (or document) for a brick based on the current root and the brick's rootMode
```

File: src/runtime/runtimeUtils.ts
Comment (mnholtz) [inline src/runtime/runtimeUtils.ts:1]: Theres a few more references to "block" in comments in this file, let's replace those too

File: src/runtime/runtimeTypes.ts [hunk 7]
```diff
@@ -15,29 +15,29 @@
...
+import { type Brick } from "@/types/brickTypes";
+import { type BrickArgs, type OutputKey } from "@/types/runtimeTypes";
 
-export type BlockType = "reader" | "effect" | "transform" | "renderer";
+export type BrickType = "reader" | "effect" | "transform" | "renderer";
 /**
- * A block configuration with the corresponding resolved IBlock and BlockType.
- * @see BlockConfig
- * @see BlockType
+ * A block configuration with the corresponding resolved Brick and BrickType.
```
Comment (mnholtz) [inline src/runtime/runtimeTypes.ts:24]:
```suggestion
 * A brick configuration with the corresponding resolved Brick and BrickType.
```

File: src/runtime/runtimeTypes.ts
Comment (mnholtz) [inline src/runtime/runtimeTypes.ts:1]: Let's also double-check this file for usages of "block" in comments

File: src/runtime/reducePipeline.ts
Comment (mnholtz) [inline src/runtime/reducePipeline.ts:1]: This file also has usages of "block" in comments

File: src/runtime/pipelineTests/pipelineTestHelpers.ts
Comment (mnholtz) [inline src/runtime/pipelineTests/pipelineTestHelpers.ts:1]: This file also has usages of the word "block" in comments

File: src/pageEditor/tabs/editTab/editTabTypes.ts [hunk 8]
```diff
@@ -19,7 +19,7 @@
  * FormikError.
  * It can be a string, a record of strings, or a record of records... i.e. it is dynamic and depends on the level of the state tree where the error happens.
  * It is never an array although we can get a nested error using index (number),
- * when the values state is represented by an array (ex. with the BlockPipeline, we'll do `PipelineErrors[0]`).
+ * when the values state is represented by an array (ex. with the BrickPipeline, we'll do `PipelineErrors[0]`).
```
Comment (mnholtz) [inline src/pageEditor/tabs/editTab/editTabTypes.ts:22]: Also change the `BlockPipeline` reference here: https://github.com/pixiebrix/pixiebrix-extension/blob/cd138b2c79c384906d75309147945a9b19d57250/src/types/runtimeTypes.ts#L98-L98

**Review by mnholtz — APPROVED** [review]
I started making some comments about the usage of the word "block" in file comments... but then I realized that there is too many of these cases to count. Unless you feel called to go through them, I'd put that out of scope fo this PR and leave for future work/a separate PR.

I would, however, go through and make sure that all explicit references in comments to actual  types, e.g. like the type in [this comment](https://github.com/pixiebrix/pixiebrix-extension/pull/5981#discussion_r1247238480), are all replaced. I found the one I mentioned in the comment by globally searching for `BrickPipeline` specifically.

File: src/pageEditor/tabs/editTab/editTabTypes.ts (hunk 8 above)
Comment (turbochef) [reply src/pageEditor/tabs/editTab/editTabTypes.ts:22]: Makes sense. I did a global search for all the items that should have been replaced and found a few. Thanks for catching it.

File: src/blocks/readers/jquery.ts (hunk 1 above)
Comment (turbochef) [reply src/blocks/readers/jquery.ts]: whoops

**Pull request merged**